dspace-api/src/main/java/org/dspace/content/clarin/ClarinItemServiceImpl.java

```diff
@@ -203,4 +210,63 @@ public void updateItemFilesMetadata(Context context, Bitstream bit) throws SQLEx
         }
         this.updateItemFilesMetadata(context, item, bundle);
     }
+
+    @Override
+    public void updateItemDatesMetadata(Context context, Item item) throws SQLException {
+        if (Objects.isNull(context)) {
+            log.error("Cannot update item dates metadata because the context is null.");
+            return;
+        }
+
+        List<MetadataValue> approximatedDates =
+                itemService.getMetadata(item, "local", "approximateDate", "issued", Item.ANY, false);
+
+        if (CollectionUtils.isEmpty(approximatedDates) || StringUtils.isBlank(approximatedDates.get(0).getValue())) {
+            log.warn("Cannot update item dates metadata because the approximate date is empty.");
```

Comment:
so this is never called on items not having approximate dates?

Reply:
because it seems that it is called on all items....

Reply:
I do not understand where is the problem. Yes this method is not called when the `...aproximateDate` is empty,